java-checks/src/main/java/org/sonar/java/checks/TextBlockTabsAndSpacesCheck.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Collections;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5664")
+public class TextBlockTabsAndSpacesCheck extends IssuableSubscriptionVisitor {
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Collections.singletonList(Tree.Kind.TEXT_BLOCK);
+  }
+
+  private static char indentationCharacter(String[] lines) {
+    for (String line : lines) {
+      if (!line.isEmpty() && isIndentation(line.charAt(0))) {
+        return line.charAt(0);
+      }
+    }
+    // If the string doesn't contain any indentation characters, it doesn't matter what we return
+    return '\0';
+  }
+
+  private static boolean isIndentation(char c) {
+    return c == ' ' || c == '\t';
+  }
+
+  private static boolean containsWrongIndentation(String[] lines) {
+    char indentationCharacter = indentationCharacter(lines);
+    for (String line : lines) {
+      for (int i = 0; i < line.length() && isIndentation(line.charAt(i)); i++) {
```

Comment:
What do you think if we consider the indentation length of the first line as indentation length for all lines in text block:
```
String s = """
....<html>
....    <body>
....        <tag>
....        </tag>
....    </body>
....</html>""";
```

Like in this example we consider only the 4 first characters in each line as indentation.

Reply:
I decided to use the actual indentation as determined by Java's rules rather than the indentation of the first line. For that I copied the code to determine that from your PR and put it into its own helper method. Depending on which PR gets merged first, the other one should be adjusted according, so the code doesn't end up duplicated.